storagenode/pieces/cache.go

```diff
@@ -0,0 +1,295 @@
+// Copyright (C) 2019 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package pieces
+
+import (
+	"context"
+	"sync"
+	"time"
+
+	"go.uber.org/zap"
+
+	"storj.io/storj/internal/sync2"
+	"storj.io/storj/pkg/storj"
+	"storj.io/storj/storage"
+)
+
+// CacheService updates the space used cache
+type CacheService struct {
+	log        *zap.Logger
+	usageCache *BlobsUsageCache
```

Comment:
It's a little unclear to me why we need both a cache and a blob cache. Could you explain this to me again?

Reply:
the cacheservice runs on a loop and syncs the cache totals to the database.  The BlobsUsageCache wraps the storage.Blobs with the actual cache